### ISSUE TITLE
sql: easier way to get a history of all versions of a descriptor

### ISSUE THREAD

**Issue by chengxiong-ruan** [cockroachdb/cockroach#107904]
With a historic view of descriptor changes over time for a schema would be super helpful for debugging potential Schema Changer bugs. We do have a few ways currently to achieve the similar goal, but they seems not straight forward or customer has chances to opt out:
(1) as @fqazi mentioned that we dump descriptor jsons in lease renewal messages, so that we could check through debugzip loggings to get older version descriptors. This might fit our goal, but it can be more explicit. Also logs has contention period. It may not last for more than a couple weeks.
(2) we have schema telemetry job which takes snapshots of schema data periodically. But it does not necessary has enough granularity and precision we would need if multiple schema changes happened within one snapshot window.

An alternative to the above is to maintain a `system.descriptor_history` table to track all versions. We may manage a contention as well, if extra disk space is a concern. But it maybe only an concern for database with a lot of descriptors.

**Comment by fqazi** [comment]
Adding to this, we can use an export request to review a descriptor's entire history when generating a debug.zip, similar to what the lease manager does.

**Comment by chengxiong-ruan** [comment]
> Adding to this, we can use an export request to review a descriptor's entire history when generating a debug.zip, similar to what the lease manager does.

Could you say more? Do we have underlying data for this already?

**Comment by fqazi** [comment]
We can take advantage of MVCC properties to go back till the GC interval and get old versions of descriptors. The lease manager already does this for as of system time queries, but in this case, it would be more general and for the entire descriptors table.

**Comment by chengxiong-ruan** [comment]
Yeah, I see what you mean. In that case we would need to be lucky enough before MVCC GC (which seems hard to predict). Or maybe have a zone config so that old versions last longer. I think, by history here, I meant "Months" at least, if not a "all time" history.
